src/main.rs

```diff
@@ -21,11 +22,53 @@ struct LockFile {
     package: Vec<Package>,
 }
 
+#[derive(Deserialize, Debug, PartialEq)]
+struct Spec {
+    op: String,
+    ver: String,
+}
+
+use std::iter::Iterator;
+
+fn parse_spec(spectxts: &str) -> Result<Vec<Spec>, String> {
+    Ok(
+        spectxts
+        .split(",")
+        .map(|item| {
+            let spectxt = item.trim();
+
+            let re = Regex::new(r"[^!=<>~]").unwrap();
+            let op = re.replace_all(spectxt, "").to_string();
+
+            let re = Regex::new(r"[^0-9\.\*]").unwrap();
+            let ver = re.replace_all(spectxt, "").to_string();
+
+            Spec { op, ver }
+        })
+        .collect()
+    )
+}
```

Comment:
```suggestion
fn parse_spec(spectxts: &str) -> Result<Vec<Spec>, String> {
        spectxts
        .split(",")
        .map(|item| {
            let spectxt = item.trim();

            let re = Regex::new(r"[^!=<>~]").unwrap();
            let op = re.replace_all(spectxt, "").to_string();

            let re = Regex::new(r"[^0-9\.\*]").unwrap();
            let ver = re.replace_all(spectxt, "").to_string();

            Spec { op, ver }
        })
        .collect()
}
```

I think we can remove the wrapping result since we unwrap it when we call the function already